src/charm.py

```diff
@@ -349,6 +395,33 @@ def _on_peer_relation_changed(self, event: RelationChangedEvent) -> None:
             self.unit.status = ActiveStatus()
             logger.debug(f"Instance {instance_label} is cluster member")
 
+    def _on_peer_relation_departed(self, event: RelationDepartedEvent) -> None:
+        """Handle the relation departed event.
+
+        Update the allowlist to remove the departing unit for all units.
```

Comment:
curious why does this need to be done for all units? What was the design motivation on this? Was it SQL based?

If its SQL based It would be nice to have a description on that here :)

Reply:
I thought that it needs to be done in all units since we need to make sure no other unit can join the group. However, I will speak with Mykola about how we need allowlists to work, and make any changes in a followup